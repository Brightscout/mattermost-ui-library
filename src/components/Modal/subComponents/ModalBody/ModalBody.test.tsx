import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

import ModalBody from '.';

// Mock props data
const modalBodyChildren = <div>{'Modal Body Children'}</div>;
const modalBodyClassName = 'modal-body-class';

describe('Modal Body', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ModalBody className={modalBodyClassName}>{modalBodyChildren}</ModalBody>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly and matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(modalBodyClassName)).toBeTruthy();
    });

    it('Should render the children correctly', () => {
        expect(component.contains(modalBodyChildren)).toBeTruthy();
    });
});
