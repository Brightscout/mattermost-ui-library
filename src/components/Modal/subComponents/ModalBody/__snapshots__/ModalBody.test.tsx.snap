// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Body Should render correctly 1`] = `
<ModalBody
  className="modal-body-class"
>
  <ModalBody
    className="modal__body padding-v-15 padding-h-12 modal-body-class"
  >
    <div
      className="modal__body padding-v-15 padding-h-12 modal-body-class modal-body"
    >
      <div>
        Modal Body Children
      </div>
    </div>
  </ModalBody>
</ModalBody>
`;
